interview-questions.md

```diff
@@ -8,36 +8,243 @@ Then, research the question to expand on your answer. Even if you feel you have
+     for the routes aspect of it rails routes-E, is very helpful. 
+     and for this specific pluralization issue I keep running into and screwing up. 
+    I have to verbally say out loud, "a student doesnt have multiple cohorts, a cohort has multiple students." 
+    I say that a few times, then shake my head like im ready to start inputting this info, but the whole time im doing it and I get to a place that may be pluralized, I have to say it outloud again just to be sure. 
+     A small mistake on the this can be very annoying. 
+     Another major thing that I have run into. 
+     way worse than anything you just did by forgetting the foreiegn key, but kind of the same. 
+     I forgot the foreign key and also had rails generate all the routes for me. 
+     and then when I went back and added the foreign key to it as described above, I realized that the routes were not working correctly and that something was now screwed up. 
+     Due to the porjects being small it is a very possible option to just clean slate restart and go again from scratch, but that will not always be an option and I think it is important that I take the time to try and figure out how to solve these errors I am running into when I make a mistake and how to resolve them instead of starting over. 
+
+
+
 2. Which RESTful routes must always be passed params? Why?
+ Your answer: 
+    A param is attached to an instance variable and the param makes it so we can pass a value from a hash. So anything that needs a specific value will absolutely need a param, i am refering to :id.
+    Show, Update, Edit, and Destroy have to have params because they need to know what to show or what to edit/update, or what to delete, and that is figured out by using the params :id. I believe we do not "have" to have them when creating. i know that we use strong params but i dont think that we have to. see example below for a visual representation of what i stated
+            For example purposes, i am going to call the model Animal
+                The RESTful routes are:
+                    Index - animals.all
+                    Show - animal/id
+                    Create - strong params
+                    New - 
+                    Update - /id
+                    Edit - /id
+                    Destroy - /id
+
 
-Your answer:
 
 Researched answer:
+I had multiple people ask me what the answer to this was and what I thought the answer was and I think that although there may be a way to create without a param, it is stupid not to. I do not know why i think you can, but you should. You should check to make sure that what a user is inputting is valid. so i am going to say, Show, Create, Update, Edit, and Destroy all need params. YOu should put the ohhhhhh wait. technically the create does not have params, it checks the vilidity in  a different method in the private section that uses params. So we will say Create needs 50% params. :)
```

Comment:
It's great that you are so comfortable that you can explain it to others. Believe it or not, explaining things differently really helps grasping the material.